src/components/Firestore/DocumentPreview/api.tsx

```diff
@@ -55,104 +55,32 @@ function adjustPayloadForArray(
   fieldPath: string[],
   value: FirestoreAny | firestore.FieldValue
 ): [firestore.FieldPath, FirestoreAny] {
-  if (fieldPath.length === 1) {
-    return [new firestore.FieldPath(...fieldPath), value];
-  }
-
-  let nestedObjectToModify: FirestoreMap | FirestoreArray = doc;
-  let topLevelArray: FirestoreArray | undefined = undefined;
-  let topLevelKey: string[] = [];
-  fieldPath.slice(0, -1).forEach((pathSegment, index) => {
-    if (isMap(nestedObjectToModify)) {
-      nestedObjectToModify = nestedObjectToModify[pathSegment] as FirestoreMap;
-    } else if (isArray(nestedObjectToModify)) {
-      nestedObjectToModify = nestedObjectToModify[
-        Number(pathSegment)
-      ] as FirestoreMap;
-    }
-    if (nestedObjectToModify instanceof Array && !topLevelArray) {
-      topLevelArray = nestedObjectToModify;
-      topLevelKey = fieldPath.slice(0, index + 1);
-    }
-  });
-
-  // If no array was found, carry on as usual
-  if (!topLevelArray) {
-    return [new firestore.FieldPath(...fieldPath), value];
-  }
-
-  const nestedFieldId = fieldPath[fieldPath.length - 1];
-  if (value !== firestore.FieldValue.delete()) {
-    // Example:
-    // {
-    //   foo: [             <-- foo == topLevelArray
-    //     {
-    //       bar: {         <-- bar == nestedObjectToModify
-    //         baz: 'test'  <-- 'baz' == nestedFieldId
-    //       }
-    //     }
-    //   ]
-    // }
-
-    if (
-      topLevelArray === nestedObjectToModify &&
-      Number(lastFieldName(fieldPath)) >= (topLevelArray as {}[]).length
-    ) {
-      // Appending a new element to the top-level array
-      return [
-        new firestore.FieldPath(...topLevelKey),
-        firestore.FieldValue.arrayUnion(value),
-      ];
+  let cur: FirestoreAny | undefined = doc;
```

Comment:
When would current be undefined?

Reply:
perhaps on a missing doc when deep navigating?

Reply:
Oh sorry, yes you're right. it doesn't look like it would be undefined ever

Reply:
`cur` will be undefined when adding a key to an object (e.g. set path `f.g` on `{"f": 1}`), with the assignment below. `doc` won't ever be `undefined` but `cur` will, thus the typing here.